haskell/haskell_haddock.bzl

```diff
@@ -243,7 +243,7 @@ def haskell_haddock_impl(ctx: AnalysisContext) -> list[Provider]:
             cmd_args(cmd, delimiter = " ", quote = "shell"),
             [
                 cmd_args(
-                    ["cp", "-f", "--reflink=auto", html, out.as_output()],
+                    ["cp", "-f", html, out.as_output()],
```

Comment:
could this cause issues previous use-cases, i.e. does anything assume that this is set? Is there a way to gate it?

Reply:
I did some digging...

This was actually introduced by commit 4edc6c19e40605602fec1fb33383fc6181cafeef which was about something not related to that topic at all.

Interestingly, there was commit 46f76f554537c49c84fc91d052df1e4ce4234bd7 which removed the `--reflink=auto` already for the same reason as this PR, but I re-introducd it accidentally in commit bc52e2fdd023cf03a1973fc7c1ff503448c225af.

Also, macOS `cp` has a flag with a similar effect: 
> `-c` copy files using clonefile(2)

We could use `uname` to check which one of the flags we can use, but it's probably not worth it. WDYT?

Reply:
Or check up-front if `cp` supports it and then use it or not?